### ISSUE TITLE
Only need Decorator

### ISSUE THREAD

**Issue by AndreasHassing** [khellang/Scrutor#142]
Hi @khellang. I'm a huge fan of what you (and your community) have done with the `Decorate` extension. It's the kind of magic I was looking for to get around ugly alternatives such as:

```csharp
services
    .AddSingleton<MyClass>()
    .AddSingleton<IMyClass, MyDecoratedClass>();

public sealed MyDecoratedClass
{
    // cannot use the interface, need to decide on a specific implementation :-(
    public MyDecoratedClass(MyClass myClass) { }
}
```

However, I would rather not add the assembly scanning component of the nuget, so I'm wondering how I should integrate just the `Decorate` code into my project.

If I copy code from your project into mine and add your MIT license to the same directory where I have the source files needed to replicate Decorate; is that good enough for you? I don't know if it makes sense, but I can also add a link to this repo at the top of the LICENSE file so it's easy to find the original creator(s)?

I'm asking to avoid any bad taste. Thank you for this project, it's fantastic 😊.

**Comment by khellang** [comment]
> If I copy code from your project into mine and add your MIT license to the same directory where I have the source files needed to replicate Decorate; is that good enough for you? I don't know if it makes sense, but I can also add a link to this repo at the top of the LICENSE file so it's easy to find the original creator(s)?

As long as you include the license- and copyright notice as mandated by the MIT license, I don't mind 😄 

It does seem a bit weird to cherry-pick methods out of NuGet libraries like this, though. What's the reasoning behind this? Do you do this with other libraries as well? 🤔

**Comment by AndreasHassing** [comment]
I know it seems weird :D that's why I had to ask. No, I have not done this with other libraries.

I don't know, this is the first time I have felt that I am trying to use a very small package that has orthogonal features (scanning and decorating) where I just want a single one of those features. So it seems feasible to slice out what I need without introducing more dependencies.

If a package was released with only the decorating feature, I would probably just use that 🤷‍♂️.
